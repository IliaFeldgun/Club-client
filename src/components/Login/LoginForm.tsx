import React from "react";
import LobbyApi from "../../api/LobbyApi";

interface ILoginFormProps {
    className: string
}
interface ILoginFormState {
    playerName: string
}
export default class LoginForm extends React.PureComponent<ILoginFormProps,ILoginFormState> {
    constructor(props: ILoginFormProps) {
        super(props)
        this.state = {playerName: ""}
        this.handleClick = this.handleClick.bind(this)
        this.handleChange = this.handleChange.bind(this)
        this.handleKeyUp = this.handleKeyUp.bind(this)
    }
    handleChange(event: React.ChangeEvent<HTMLInputElement>) {
        this.setState({
            playerName: event.target.value
        })
    }
    handleClick(event: React.MouseEvent<HTMLButtonElement>) {
        this.sendLogin()
    }
    handleKeyUp(event: React.KeyboardEvent<HTMLInputElement>) {
        if (event.keyCode === 13) { // "RETURN" key
            this.sendLogin()
        }
    }
    sendLogin() {
        if(this.state.playerName) {
            LobbyApi.newPlayer(this.state.playerName).then((isCreated) => {
                window.location.reload()
            })
        }
    }
    render() {
        const fieldClass = "form-field"
        const allClass = this.props.className + " login-form"
        const buttonClass = "form-button"
        return (
            <React.Fragment>
                <div className={allClass}>
                    <h3>Login with a name of your choice</h3>
                    <input 
                        className={fieldClass} 
                        id="playerName" 
                        type="text" 
                        name="playerName" 
                        onChange={this.handleChange} 
                        onKeyUp={this.handleKeyUp} 
                    />
                    <button className={buttonClass} id="loginsend" onClick={this.handleClick}>
                        <span>
                            Login
                        </span>
                    </button>
                </div>
            </React.Fragment>
        )
    }
}