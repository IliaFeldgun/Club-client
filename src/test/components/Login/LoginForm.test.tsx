import React from 'react'
import LoginForm from '../../../components/Login/LoginForm'
import { mount, render, shallow } from 'enzyme'

describe('Testing LoginForm', () => {
    test('component renders', () => {
        const component = shallow(<LoginForm />)
        expect(component).toMatchSnapshot()
    })
})