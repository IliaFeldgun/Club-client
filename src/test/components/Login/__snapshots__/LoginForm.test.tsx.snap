// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LoginForm Component renders without props 1`] = `
<Fragment>
  <div
    className=" login-form"
  >
    <h3>
      Login with a name of your choice
    </h3>
    <input
      className="form-field"
      id="playerName"
      name="playerName"
      onChange={[Function]}
      onKeyUp={[Function]}
      type="text"
    />
    <button
      className="form-button"
      id="loginsend"
      onClick={[Function]}
    >
      <span>
        Login
      </span>
    </button>
  </div>
</Fragment>
`;

exports[`Testing LoginForm component renders 1`] = `
<Fragment>
  <div
    className="form test login-form"
  >
    <h3>
      Login with a name of your choice
    </h3>
    <input
      className="form-field"
      id="playerName"
      name="playerName"
      onChange={[Function]}
      onKeyUp={[Function]}
      type="text"
    />
    <button
      className="form-button"
      id="loginsend"
      onClick={[Function]}
    >
      <span>
        Login
      </span>
    </button>
  </div>
</Fragment>
`;
